### PR TITLE
fix some cppcheck and compiler warnings

### PR DESCRIPTION
**Pull request by matthiaskrgr**

### CODE REVIEW

File: main.cpp [hunk 1]
```diff
@@ -57,11 +57,11 @@ int main(int argc, char **argv)
     std::map<std::string, simplecpp::TokenList*> included = simplecpp::load(rawtokens, files, dui, &outputList);
     for (std::pair<std::string, simplecpp::TokenList *> i : included)
         i.second->removeComments();
-    simplecpp::TokenList output(files);
-    simplecpp::preprocess(output, rawtokens, files, included, dui, &outputList);
+    simplecpp::TokenList tokenList(files);
```
Comment (danmar) [inline main.cpp]: maybe outputTokens or something like that is better

**Comment by danmar** [comment]
I don't know.. I have sympathy for Wshadow in general, it can prevent mistakes, but often we end up with less good names. maybe it's time to remove it from simplecpp/cppcheck and maybe just run it occasionally to make sure there are not mistakes.

File: main.cpp (hunk 1 above)
Comment (matthiaskrgr) [inline main.cpp]: Ok, I changed it to outputTokens.

**Comment by danmar** [comment]
ok well.. I'll push this.. but I am thinking that Wshadow should rather be removed.

**Pull request merged**